mne/decoding/time_gen.py

```diff
@@ -914,3 +953,358 @@ def time_generalization(epochs_list, clf=None, cv=5, scoring="roc_auc",
+        """
+        from scipy import stats
+        n_jobs = self.n_jobs
+
+        if not hasattr(self, 'estimators_'):
+            raise ValueError("Estimators have to be fit before predicting.")
+
+        if picks is None:
+            picks = pick_types(epochs.info, meg=True, eeg=True,
+                               exclude='bads')
+
+        event_ids = epochs.event_id
+        epochs_list = [epochs[event] for event in event_ids]
+        equalize_epoch_counts(epochs_list)
+        X = np.concatenate(
+            [e.get_data()[:, picks, :] for e in epochs_list])
+
+        if len(X) == len(self.estimators_):
+            raise ValueError("Number of time points should be consistent with "
+                             "the fit data.")
+
+        def _predict(estimator_folds, X):
+            n_epochs = X.shape[0]
+            n_cv = len(estimator_folds)
+            n_classes = len(estimator_folds[0].classes_)
+
+            # Initialization
+            if self.predict_mode == "mean-prediction":
+                if self.predict_type == "predict_proba":
+                    predicted = np.zeros((n_epochs, n_classes, n_cv))
+                else:
+                    predicted = np.zeros((n_epochs, 1, n_cv))
+
+            else:
+                if self.predict_type == "predict_proba":
+                    predicted = np.zeros((n_epochs, n_classes))
+                else:
+                    predicted = np.zeros((n_epochs, 1))
+
+            if self.predict_mode == "mean-prediction":
+                for i, estimator in enumerate(estimator_folds):
+                    if self.predict_type == "predict":
+                        predicted[:, :, i] = np.reshape(
+                            estimator.predict(X), (-1, 1))
+                    elif self.predict_type == "predict_proba":
+                        predicted[:, :, i] = estimator.predict_proba(X)
+                    else:
+                        predicted[:, :, i] = np.reshape(
+                            estimator.decision_function(X), (-1, 1))
+
+                if self.predict_type == "predict":
+                    return stats.mode(predicted, axis=2)[0][:, :, 0]
+                else:
+                    return np.mean(predicted, axis=2)
+
+            if self.predict_mode == "cross-validation":
+                for i, (train, test) in enumerate(self.cv_):
+                    estimator = estimator_folds[i]
+                    if self.predict_type == "predict":
+                        predicted[test, :] = estimator.predict(
+                            X[test, :])[:, None]
+                    elif self.predict_type == "predict_proba":
+                        predicted[test, :] = estimator.predict_proba(
+                            X[test, :])
+                    else:
+                        predicted[test, :] = estimator.decision_function(
+                            X[test, :])[:, None]
+                return predicted
```

Comment:
I think we meanwhile agreed on not adding nested functions unless we really need to.